### DISCUSSION TITLE
SHARPy Installation Problems - Could not find MKL

### DISCUSSION THREAD

**Discussion by askerp** [ImperialCollegeLondon/sharpy#129] [Q&A] [answered]
I'm trying to install SHARPy on my virtual machine (Oracle VM Virtualbox), which has Linux-Ubuntu 18.04 as OS. As per the instructions on the website (https://ic-sharpy.readthedocs.io/en/master/content/installation.html), I was able to install all the prerequisites (software/package) such as HDFview, python3.8, anaconda, paraview, gcc, gfortran,git etc. 
In the quick install, I was successful till the 3rd step which created the 'build' directory. After this when I enter "cmake .." into the terminal window, I get the following error:

_-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The Fortran compiler identification is GNU 7.5.0
-- Check for working Fortran compiler: /usr/bin/gfortran
-- Check for working Fortran compiler: /usr/bin/gfortran  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /usr/bin/gfortran supports Fortran 90
-- Checking whether /usr/bin/gfortran supports Fortran 90 -- yes
-- Could NOT find MKL (missing: MKL_LIBRARIES MKL_INCLUDE_DIRS MKL_INTERFACE_LIBRARY MKL_SEQUENTIAL_LAYER_LIBRARY MKL_CORE_LIBRARY) 
-- Looking for Fortran sgemm
-- Looking for Fortran sgemm - not found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Looking for Fortran sgemm
-- Looking for Fortran sgemm - found
-- Found BLAS: /usr/lib/x86_64-linux-gnu/libblas.so  
-- Looking for Fortran cheev
-- Looking for Fortran cheev - found
-- A library with LAPACK API found.
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP_Fortran: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- Could NOT find MKL (missing: MKL_LIBRARIES MKL_INCLUDE_DIRS MKL_INTERFACE_LIBRARY MKL_SEQUENTIAL_LAYER_LIBRARY MKL_CORE_LIBRARY) 
-- A library with LAPACK API found.
-- Found OpenMP_C: -fopenmp  
-- Found OpenMP_CXX: -fopenmp  
-- Found OpenMP: TRUE   
-- Configuring done
-- Generating done
-- Build files have been written to: /home/askerp/sharpy/build_

When I try to write "cmake .." again it shows this:

_-- Could NOT find MKL (missing: MKL_LIBRARIES MKL_INCLUDE_DIRS MKL_INTERFACE_LIBRARY MKL_SEQUENTIAL_LAYER_LIBRARY MKL_CORE_LIBRARY) 
-- A library with LAPACK API found.
-- Found OpenMP_Fortran: -fopenmp  
-- Found OpenMP: TRUE   
-- Could NOT find MKL (missing: MKL_LIBRARIES MKL_INCLUDE_DIRS MKL_INTERFACE_LIBRARY MKL_SEQUENTIAL_LAYER_LIBRARY MKL_CORE_LIBRARY) 
-- A library with LAPACK API found.
-- Found OpenMP: TRUE   
-- Configuring done
-- Generating done
-- Build files have been written to: /home/askerp/sharpy/build_

I have tried to install MKL separately but this error still keeps coming up whenever I run the cmake command.

**Comment by ArturoMS13** [comment, marked as answer]
Hi askerp,

I believe you can ignore this error about the MKL library. 

Can you try these steps and come back to us with the output?
- Remember to activate the sharpy environment (``conda activate sharpy_env``) 
- Run ``cmake ..`` and then ``make install`` in the build directory

Best regards, Arturo.

**Reply by askerp** [reply]
I tried the steps you asked and the output is as shown in the image below,

![Screenshot (1194)](https://user-images.githubusercontent.com/74264453/111866444-625e9500-8965-11eb-80c4-393f8b92315f.png)

**Reply by ArturoMS13** [reply]
Nice! SHARPy must be working now

**Reply by askerp** [reply]
Thank you, yeah it is working now.
